src/Kernel/client/ExecutionPathVisualizer/utils.ts

```diff
@@ -1,6 +1,18 @@
 import { Metadata } from './metadata';
 import { GateType, minGateWidth, labelPadding, labelFontSize, argsFontSize } from './constants';
 
+/**
+ * Generate a UUID using `Math.random`.
+ *
+ * @returns UUID string.
+ */
+const createUUID = (): string =>
+    'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+        const r = (Math.random() * 16) | 0,
+            v = c == 'x' ? r : (r & 0x3) | 0x8;
+        return v.toString(16);
+    });
```

Comment:
It may be worth adding a link to where this implementation came from ([this post](https://stackoverflow.com/questions/105034/how-to-create-guid-uuid)? 😊) and a note that it's not cryptographically secure. (Of course, it's good enough for this application!)

Reply:
That's a good point, thanks!